dev/com.ibm.ws.logging/src/com/ibm/ws/logging/source/TraceSource.java

```diff
@@ -86,33 +88,48 @@ public String getLocation() {
     public void publish(RoutedMessage routedMessage) {
         //Publish the message if it is not coming from a handler thread
         if (!ThreadLocalHandler.get()) {
-            LogRecord logRecord = routedMessage.getLogRecord();
-            if (logRecord != null) {
-                if (bufferMgr != null) {
-                    bufferMgr.add(parse(routedMessage, logRecord));
-                }
+            if (routedMessage.getLogRecord() != null && bufferMgr != null) {
+                bufferMgr.add(parse(routedMessage));
             }
         }
     }
 
-    public TraceLogData parse(RoutedMessage routedMessage, LogRecord logRecord) {
-        String message = routedMessage.getFormattedVerboseMsg();
-        if (message == null)
-            message = logRecord.getMessage();
-        long timestamp = logRecord.getMillis();
-        int threadId = logRecord.getThreadID();
-        String loggerName = logRecord.getLoggerName();
-        String logLevel = LogFormatUtils.mapLevelToType(logRecord);
-        String logLevelRaw = LogFormatUtils.mapLevelToRawType(logRecord);
-        String methodName = logRecord.getSourceMethodName();
-        String className = logRecord.getSourceClassName();
-        Map<String, String> extensions = null;
-        if (logRecord instanceof WsLogRecord)
-            extensions = ((WsLogRecord) logRecord).getExtensions();
-        String sequence = sequenceNumber.next(timestamp);
-        //String sequence = timestamp + "_" + String.format("%013X", seq.incrementAndGet());
-
-        return new TraceLogData(timestamp, threadId, loggerName, logLevel, logLevelRaw, message, methodName, className, extensions, sequence);
+    public GenericData parse(RoutedMessage routedMessage) {
+
+        GenericData genData = new GenericData();
+        LogRecord logRecord = routedMessage.getLogRecord();
+        String verboseMessage = routedMessage.getFormattedVerboseMsg();
+        if (verboseMessage == null) {
+            genData.addPair("message", logRecord.getMessage());
+        } else {
+            genData.addPair("message", verboseMessage);
+        }
+
+        long datetimeValue = logRecord.getMillis();
+        // Must pass datetime as string, as its value type must be set as string in its kvp
+        genData.addPair("ibm_datetime", Long.toString(datetimeValue));
```

Comment:
just add ibm_datetime as a long dateVal... it's formatter's job to decide if we want it to be a formatted like a string.